runtime/bcutil/cfreader.c

```diff
@@ -3470,15 +3481,30 @@ readAnnotationElement(J9CfrClassFile * classfile, J9CfrAnnotationElement ** pAnn
 		break;
 			
 	case 'e':
+	{
+		J9CfrConstantPoolInfo* cpBase = classfile->constantPool;
+		U_16 cpCount = classfile->constantPoolCount;
+		U_16 typeNameIndex = 0;
+		U_16 constNameIndex = 0;
+
 		if (!ALLOC_CAST(element, J9CfrAnnotationElementEnum, J9CfrAnnotationElement)) {
 			return -2;
 		}
 
 		CHECK_EOF(4);
 		NEXT_U16(((J9CfrAnnotationElementEnum *)element)->typeNameIndex, index);
 		NEXT_U16(((J9CfrAnnotationElementEnum *)element)->constNameIndex, index);
+		
+		typeNameIndex = ((J9CfrAnnotationElementEnum *)element)->typeNameIndex;
+		constNameIndex = ((J9CfrAnnotationElementEnum *)element)->constNameIndex;
+		if(!typeNameIndex || (typeNameIndex >= cpCount) || !constNameIndex || (constNameIndex >= cpCount)) {
```

Comment:
Please format these properly (space after if) and don't use ! for zero tests in new code.

Reply:
Fixed as suggested above.